src/lib/Microsoft.Health.Extensions.Fhir.R4/FhirClientFactory.cs

```diff
@@ -64,18 +65,29 @@ public FhirClient Create()
 
         private static FhirClient CreateClient(TokenCredential tokenCredential, ITelemetryLogger logger)
         {
+            EnsureArg.IsNotNull(tokenCredential, nameof(tokenCredential));
+
             var url = Environment.GetEnvironmentVariable("FhirService:Url");
             EnsureArg.IsNotNullOrEmpty(url, nameof(url));
             var uri = new Uri(url);
 
-            EnsureArg.IsNotNull(tokenCredential, nameof(tokenCredential));
-
             var fhirClientSettings = new FhirClientSettings
             {
                 PreferredFormat = ResourceFormat.Json,
             };
 
-            var client = new FhirClient(url, fhirClientSettings, new BearerTokenAuthorizationMessageHandler(uri, tokenCredential, logger));
+            FhirClient client;
+            try
+            {
+                client = new FhirClient(url, fhirClientSettings, new BearerTokenAuthorizationMessageHandler(uri, tokenCredential, logger));
+            }
+            catch (Exception ex)
+            {
+                client = null;

```

Comment:
Do we actually get back a certain value for `client` in case of such an exception?

Reply:
`BaseFhirClient` throws `ArgumentNullException` or `ArgumentException` if the endpoint is invalid.

In commit 9ab3837, I've initialized `client` to null.